augur/filter.py

```diff
@@ -1602,10 +1602,20 @@ def run(args):
             observed_sequence_strains = set()
             with open_file(args.output, "wt") as output_handle:
                 for sequence in sequences:
-                    observed_sequence_strains.add(sequence.id)
-
                     if sequence.id in valid_strains:
-                        write_sequences(sequence, output_handle, 'fasta')
+                        # Do not emit sequences for strains we've already
+                        # observed. These duplicates should likely be addressed
+                        # upstream, but at least warn the user about specific
+                        # duplicates to support debugging.
+                        if sequence.id in observed_sequence_strains:
+                            print(
+                                f"WARNING: Found duplicate sequence for strain '{sequence.id}'.",
+                                "This sequence will not be in the output sequences.",
```

Comment:
I misinterpreted this as all duplicates will be dropped. I think it would be helpful to explicitly say "Only the first sequence will be included in the output sequences".

Reply:
Good call. I just rebased and pushed a version with this fix (b7f3108).